### PR TITLE
[added] Allow overlay override

### PR DESCRIPTION
**Pull request by cema-sp**
Continues and Closes #281

@diasbruno @claydiffrient Sorry that it took so long for me to prepare this feature. Leaving [previous branch](cema-sp/react-modal) if someone uses it.

To override overlay one should provide `overlay` prop to `Modal` containing react element. That component will be cloned and filled with overlay props and children (using `React.cloneElement`).

Another way may be to provide `overlayComponent` and `overlayProps` and then create react element from them (using `React.createElement`). Which way us better and more idiomatic?

Changes proposed:
- Extract `ModalOverlay` to component
- Allow user to provide custom overlay with `Modal.props.overlay` prop

Acceptance Checklist:
- [x] All commits have been squashed to one.
- [x] The commit message follows the guidelines in `CONTRIBUTING.md`.
- [x] Documentation (README.md) and examples have been updated as needed.
- [x] If this is a code change, a spec testing the functionality has been added.

### CODE REVIEW

**Comment by diasbruno** [comment]
@cema-sp this looks awesome. i'll bring some feedback later.

**Comment by cema-sp** [comment]
@diasbruno Rebased & squashed

File: examples/basic/app.js [hunk 1]
```diff
@@ -9,25 +9,34 @@ Modal.setAppElement('#example');
 var App = React.createClass({
 
   getInitialState: function() {
-    return { modalIsOpen: false };
+    return { modalIsOpen: false, modal2: false };
   },
 
   openModal: function() {
-    this.setState({modalIsOpen: true});
+    this.setState({ ...this.state, modalIsOpen: true });
```
Comment (lindskogen) [inline examples/basic/app.js]: Is there some special reason for this? `setState` actually does shallow merge automatically.

**Comment by Monte9** [comment]
@diasbruno would this PR get merged at any point soon? Would love to have the scroll functionality on the modal. thanks.

**Comment by rifkegribenes** [comment]
@Monte9 FYI if all you need is a scrollable modal, you can just add
```
.ReactModal__Overlay {
  overflow-y: scroll !important;
 }
```
to your css

**Comment by diasbruno** [comment]
@rifkegribenes that's correct. the idea for this PR is to make the overlay customizable.

@cema-sp I think it's a good time to 'revive' this branch. Any help you need to get this branch updated, please let me know.

**Comment by hszeto** [comment]
@rifkegribenes thanks for the quick css fix.

I found the following works better for my case. It retains the modal positon. (I have the modal open a little closer to the top instead of center.)

````
.ReactModal__Overlay--after-open {
  overflow-y: scroll !important;
}

**Comment by cema-sp** [comment]
Sorry, won't be able to get back to it any time soon.

**Pull request closed without merging**